src/licensor.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2016 Open Permissions Platform Coalition
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+import defaultsDeep from 'lodash.defaultsdeep';
+import _get from 'lodash.get';
+import pickBy from 'lodash.pickby';
+import jsonld from 'jsonld';
+import 'isomorphic-fetch';
+
+import {names} from './constants';
+
+/**
+ * Extract relevant information and fetch the organisation
+ */
+function transformLicensor(data, options) {
+  let licensor = defaultsDeep({}, data.licensor, options);
+  //Get navigation link
+  let link = _get(licensor, ['reference_links', 'links', data.source_id_type]);
+  if (link) {
+    link = link.replace(/{source_id}/g, data.source_id);
+    link = link.replace(/{source_id_type}/g, data.source_id_type);
+    licensor.link = link;
+  } else if (licensor.website) {
+    licensor.link = licensor.website;
+  }
+  return licensor;
+}
+
+export default {
+  /**
+   * Parse a source id/source id type to get licensor information
+   */
+  parseLicensor: function (data, options={}) {
+    const url = `${options.query}/licensors?source_id=${data.source_id}&source_id_type=${data.source_id_type}`;
+    return fetch(url)
+      .then(response => {
```

Comment:
I see, you should still move the HTTP status check to before `response.json` because some HTTP responses won't have JSON (e.g. if there is a connection error)